packages/server/src/automations/automationUtils.ts

```diff
@@ -96,6 +98,93 @@ export function getError(err: any) {
   return typeof err !== "string" ? err.toString() : err
 }
 
+export async function sendAutomationAttachmentsToStorage(
+  tableId: string,
+  row: Row
+) {
+  let table = await sdk.tables.getTable(tableId)
+  const attachmentRows: Record<string, AutomationAttachment[]> = {}
+
+  for (const [prop, value] of Object.entries(row)) {
+    const schema = table.schema[prop]
+    if (
+      schema?.type === FieldType.ATTACHMENTS ||
+      schema?.type === FieldType.ATTACHMENT_SINGLE
+    ) {
+      attachmentRows[prop] = value
+    }
+  }
+
+  for (const [prop, attachments] of Object.entries(attachmentRows)) {
+    if (attachments.length) {
+      row[prop] = await Promise.all(
+        attachments.map(attachment => generateAttachmentRow(attachment))
+      )
+    }
+  }
+
+  return row
+}
+
+async function generateAttachmentRow(attachment: AutomationAttachment) {
+  const prodAppId = context.getProdAppId()
+  try {
+    let size: number | null | undefined
+    let s3Key: string
+    const client = objectStore.ObjectStore(objectStore.ObjectStoreBuckets.APPS)
+
+    const attachmentResult = await objectStore.processAutomationAttachment(
+      attachment
+    )
+    const extension = attachment.filename.split(".").pop() || ""
+
+    if ("bucket" in attachmentResult && "path" in attachmentResult) {
+      const { path, content, bucket } = attachmentResult
+      let client = objectStore.ObjectStore(bucket)
+
+      if (path.includes(`${prodAppId}/attachments/`)) {
+        s3Key = path
+      } else {
+        const processedFileName = `${uuid.v4()}.${extension}`
+        s3Key = `${prodAppId}/attachments/${processedFileName}`
+        await objectStore.streamUpload({
+          bucket: objectStore.ObjectStoreBuckets.APPS,
+          stream: content,
+          filename: s3Key,
+        })
+      }
+      const metadata = await client
```

Comment:
Can these calls be encapsulated in the `objectStore` rather than exporting the Client itself and running this logic directly here? 

Like an `objectStore.exists(bucket, path)` would be quite useful and would avoid the need for setting up the client etc here.

Reply:
How's that look now?